src/lib/routers/history.ts

```diff
@@ -119,7 +120,8 @@ class BrowserHistory implements Router {
   /**
    * Pushes a search state into the URL.
    */
-  public write(routeState: RouteState): void {
+  public write(routeState: RouteState, replace: boolean = false): void {
+    if (this._disposed) return;
```

Comment:
if you don't write when the dispose has happened, will the replaceState even ever happen?

Reply:
yes, because the `_dispose` is set after the final `write` call.
You can tell it runs by the looking [at the test](https://github.com/algolia/instantsearch.js/pull/4808/files#diff-b6b548e96fccd4f7c292cf53d63f1ae7314c42cad4c51e3b354f985d438b3203R580-R586): the URL is indeed updated, the history stack didn't change length.